File: scripts/seed-magnet-words.ts
import { createClient } from "@supabase/supabase-js";

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL!,
  process.env.SUPABASE_SERVICE_ROLE_KEY!
);

const words = [
  // ─── S-TIER ───
  { word: "secretly", grade: "S", category: "Intrigue", lift_range: "+38-52%", why_it_works: "Implies hidden knowledge most people don't have access to — triggers exclusivity and FOMO instantly.", example_before: "How creators grow on YouTube", example_after: "how creators are secretly doubling their views", niches: [] },
  { word: "actually", grade: "S", category: "Reframe", lift_range: "+35-48%", why_it_works: "Signals the viewer's current belief is wrong — creates a curiosity gap that demands resolution.", example_before: "AI tools for YouTube", example_after: "what AI tools actually do to your retention", niches: [] },
  { word: "nobody", grade: "S", category: "Exclusivity", lift_range: "+40-55%", why_it_works: "Implies the viewer is about to access information that the crowd doesn't have. Pure exclusivity trigger.", example_before: "Growing a channel from zero", example_after: "what nobody tells you about growing from zero", niches: [] },
  { word: "quietly", grade: "S", category: "Stealth", lift_range: "+36-50%", why_it_works: "Implies something significant is happening under the radar — creates urgency to catch up before everyone else.", example_before: "Algorithm changes in 2024", example_after: "why the algorithm is quietly punishing this mistake", niches: [] },
  { word: "hidden", grade: "S", category: "Discovery", lift_range: "+38-54%", why_it_works: "Discovery framing — implies there's a layer beneath what everyone else can see.", example_before: "YouTube monetization tips", example_after: "the hidden reason your monetization is lower than expected", niches: [] },
  { word: "exposed", grade: "S", category: "Revelation", lift_range: "+42-58%", why_it_works: "Controversy + revelation combined. Feels like something previously suppressed is being uncovered.", example_before: "The truth about faceless channels", example_after: "faceless channels exposed — what they don't show you", niches: [] },
  { word: "truth", grade: "S", category: "Authenticity", lift_range: "+33-47%", why_it_works: "Signals that everything the viewer has been told before was false or incomplete.", example_before: "Building an audience online", example_after: "the truth about building an audience nobody wants to say", niches: [] },
  { word: "revealed", grade: "S", category: "Disclosure", lift_range: "+35-49%", why_it_works: "Passive framing that makes the viewer feel they're receiving privileged information being unlocked for them.", example_before: "Viral video formula", example_after: "the viral video formula finally revealed (it's not what you think)", niches: [] },
  // ─── A-TIER ───
  { word: "destroyed", grade: "A", category: "Stakes", lift_range: "+28-40%", why_it_works: "High-stakes outcome language — implies serious consequence the viewer needs to avoid or understand.", example_before: "Common YouTube mistakes", example_after: "the one mistake that destroyed a 100k channel overnight", niches: [] },
  { word: "ruined", grade: "A", category: "Consequence", lift_range: "+26-38%", why_it_works: "Consequence framing creates empathy and warning — viewer wants to avoid the same fate.", example_before: "Niche selection mistakes", example_after: "how picking the wrong niche ruined 3 years of work", niches: [] },
  { word: "wrong", grade: "A", category: "Correction", lift_range: "+25-37%", why_it_works: "Everyone fears being wrong. Signals that a common belief the viewer holds is incorrect.", example_before: "Growing with YouTube Shorts", example_after: "everything you know about Shorts is wrong", niches: [] },
  { word: "real", grade: "A", category: "Authenticity", lift_range: "+24-36%", why_it_works: "Cuts through filtered content. Viewers crave unfiltered reality — 'real' promises they'll get it.", example_before: "YouTube income at 100k subscribers", example_after: "the real YouTube income at 100k (not the highlight reel)", niches: [] },
  { word: "dangerous", grade: "A", category: "Warning", lift_range: "+27-39%", why_it_works: "Warns of a risk the viewer is likely unaware of — creates urgency to watch immediately.", example_before: "Outsourcing your YouTube content", example_after: "why outsourcing your YouTube content is more dangerous than you think", niches: [] },
  { word: "mistake", grade: "A", category: "Learning", lift_range: "+24-35%", why_it_works: "Mistake framing is highly relatable — most creators fear making errors they don't know about.", example_before: "Building a YouTube channel", example_after: "the silent mistake killing most YouTube channels in month 3", niches: [] },
  { word: "trap", grade: "A", category: "Warning", lift_range: "+26-38%", why_it_works: "Implies the viewer is being manipulated or misled without knowing it — triggers protective instinct.", example_before: "Viral video strategies", example_after: "the viral strategy trap most creators fall into", niches: [] },
  { word: "brutal", grade: "A", category: "Honesty", lift_range: "+23-34%", why_it_works: "Promises radical transparency. Viewers are tired of sugar-coated advice — brutal implies real talk.", example_before: "Honest YouTube growth advice", example_after: "brutal truth about YouTube growth in 2024", niches: [] },
  { word: "fake", grade: "A", category: "Authenticity", lift_range: "+25-36%", why_it_works: "Exposes something fraudulent — creates distrust of conventional wisdom and makes the creator the trustworthy alternative.", example_before: "YouTube growth strategies that work", example_after: "the fake YouTube growth strategies everyone keeps recommending", niches: [] },
  { word: "silent", grade: "A", category: "Hidden Force", lift_range: "+24-35%", why_it_works: "Something operating in the background, unseen — implies the viewer is affected without knowing it.", example_before: "Things hurting your channel", example_after: "the silent algorithm change hurting small channels right now", niches: [] },
  { word: "killing", grade: "A", category: "Impact", lift_range: "+28-41%", why_it_works: "Strong outcome language — implies something is causing serious, ongoing damage to the viewer's results.", example_before: "Why videos underperform", example_after: "the one habit quietly killing your video performance", niches: [] },
  { word: "broken", grade: "A", category: "Problem", lift_range: "+22-33%", why_it_works: "Diagnoses a system the viewer relies on as fundamentally flawed — creates urgency to find the fix.", example_before: "YouTube advice that doesn't work", example_after: "why the standard YouTube advice is completely broken", niches: [] },
  // ─── B-TIER ───
  { word: "ignored", grade: "B", category: "Overlooked", lift_range: "+18-28%", why_it_works: "Implies the viewer has been leaving value on the table by not noticing something obvious.", example_before: "YouTube analytics features", example_after: "the ignored analytics feature that predicts viral videos", niches: [] },
  { word: "missed", grade: "B", category: "FOMO", lift_range: "+17-27%", why_it_works: "Fear of missing out — viewer feels they've already lost something and need to catch up.", example_before: "YouTube growth opportunities", example_after: "the growth window most creators completely missed", niches: [] },
  { word: "banned", grade: "B", category: "Forbidden", lift_range: "+20-30%", why_it_works: "Forbidden knowledge framing — implies the information is being suppressed or restricted.", example_before: "YouTube monetization strategies", example_after: "the monetization strategy YouTube quietly banned creators from using", niches: [] },
  { word: "toxic", grade: "B", category: "Warning", lift_range: "+18-27%", why_it_works: "Strong negative framing that implies ongoing harm — creates urgency to identify and eliminate the problem.", example_before: "Common YouTube habits", example_after: "5 toxic YouTube habits creators defend but secretly regret", niches: [] },
  { word: "outdated", grade: "B", category: "Timely", lift_range: "+16-25%", why_it_works: "Signals that the viewer's current strategy is no longer working — creates urgency to update.", example_before: "YouTube growth tactics", example_after: "the outdated YouTube tactics that are now hurting your channel", niches: [] },
  { word: "overrated", grade: "B", category: "Contrarian", lift_range: "+17-26%", why_it_works: "Contrarian framing — challenges popular opinion and signals the creator has a better perspective.", example_before: "Viral video advice", example_after: "the most overrated viral video advice creators keep following", niches: [] },
  { word: "underrated", grade: "B", category: "Discovery", lift_range: "+16-24%", why_it_works: "Discovery framing — implies the viewer will learn about something valuable most people dismiss.", example_before: "YouTube growth strategies", example_after: "the most underrated YouTube growth strategy in 2024", niches: [] },
  { word: "uncomfortable", grade: "B", category: "Honest", lift_range: "+15-24%", why_it_works: "Implies the creator is going to say something most people avoid saying — raw honesty hook.", example_before: "The creator economy reality", example_after: "the uncomfortable reality of the creator economy nobody discusses", niches: [] },
  { word: "strange", grade: "B", category: "Curiosity", lift_range: "+16-25%", why_it_works: "Pattern interrupt — implies something unusual or counterintuitive about a familiar topic.", example_before: "What makes videos go viral", example_after: "the strange reason some videos go viral 6 months after posting", niches: [] },
  { word: "dark", grade: "B", category: "Intrigue", lift_range: "+17-26%", why_it_works: "Implies the creator is going to expose the negative side of something the viewer participates in.", example_before: "Behind the scenes of viral channels", example_after: "the dark side of viral YouTube channels nobody admits", niches: [] },
  // ─── C-TIER ───
  { word: "proven", grade: "C", category: "Credibility", lift_range: "+10-18%", why_it_works: "Adds social proof and reduces viewer skepticism — implies the method has been validated.", example_before: "YouTube growth system", example_after: "the proven YouTube growth system behind 50+ channels", niches: [] },
  { word: "simple", grade: "C", category: "Accessibility", lift_range: "+9-16%", why_it_works: "Reduces perceived barrier to entry — viewer believes they can execute the strategy.", example_before: "Getting your first 1000 subscribers", example_after: "the simple system behind getting your first 1000 subscribers", niches: [] },
  { word: "exact", grade: "C", category: "Specificity", lift_range: "+11-19%", why_it_works: "Precision framing — viewer expects a concrete, actionable template rather than vague advice.", example_before: "Video script formula", example_after: "the exact script formula behind 10M-view videos", niches: [] },
  { word: "weird", grade: "C", category: "Curiosity", lift_range: "+12-20%", why_it_works: "Pattern interrupt — signals something counterintuitive or unexpected is about to be shared.", example_before: "Retention trick for videos", example_after: "the weird retention trick that keeps viewers to the end", niches: [] },
  { word: "untold", grade: "C", category: "Exclusive", lift_range: "+13-21%", why_it_works: "Implies suppressed or overlooked narrative — viewer feels they're getting the version nobody else has.", example_before: "Story of a viral channel", example_after: "the untold story of how this channel went from 0 to 1M", niches: [] },
];

async function seed() {
  console.log("Seeding", words.length, "magnet words...");
  const { data, error } = await supabase
    .from("magnet_words")
    .upsert(
      words.map(w => ({ ...w, is_active: true })),
      { onConflict: "word" }
    );
  if (error) {
    console.error("Error:", error.message);
  } else {
    console.log("Done! Seeded", words.length, "words.");
  }
}

seed();

File: scripts/seed_magnet_words.py
#!/usr/bin/env python3
"""Quality expansion of magnet_words — hand-vetted, niche-tagged, grade-matched.
No scraped junk, no fabricated per-word multipliers (lift_range follows the
existing grade bands already in the table). Dedupes against current words.
Run: python3 scripts/seed_magnet_words.py
"""
import os, json, urllib.request, urllib.error

env = {}
for line in open(os.path.join(os.path.dirname(__file__), "..", ".env.local")):
    line = line.strip()
    if "=" in line and not line.startswith("#"):
        k, v = line.split("=", 1); env[k.strip()] = v.strip().strip('"').strip("'")
URL = env["NEXT_PUBLIC_SUPABASE_URL"]; KEY = env["SUPABASE_SERVICE_ROLE_KEY"]
H = {"apikey": KEY, "Authorization": f"Bearer {KEY}", "Content-Type": "application/json"}

# Grade lift bands match the existing table convention (rotated for natural variation)
BANDS = {
    "S": ["+40-55%", "+38-52%", "+42-58%", "+36-50%"],
    "A": ["+26-38%", "+28-40%", "+24-36%", "+30-42%"],
    "B": ["+16-26%", "+18-28%", "+15-24%", "+17-27%"],
    "C": ["+9-16%", "+10-18%", "+11-19%", "+8-15%"],
}

# (word, grade, category, [niche ids], why_it_works, example_before, example_after)
W = [
    # ── S-tier: maximum-pull curiosity / shock / stakes ──────────────────────
    ("insane","S","Shock",[], "Signals the content breaks normal expectation, triggering an involuntary double-take.", "How Compound Interest Works", "How Compound Interest Gets Insane"),
    ("impossible","S","Curiosity",[], "Frames the topic as something that shouldn't be able to happen, demanding explanation.", "Surviving in Deep Space", "The Impossible Way to Survive Deep Space"),
    ("terrifying","S","Fear",[], "Raises emotional stakes instantly; fear is the strongest scroll-stopping trigger.", "What Sharks Really Do", "The Terrifying Truth About What Sharks Do"),
    ("deadly","S","Stakes",["fitness","science","true-crime","cooking"], "Attaches life-or-death stakes to an everyday topic.", "Common Kitchen Mistakes", "The Deadly Kitchen Mistake Everyone Makes"),
    ("rigged","S","Conspiracy",["personal-finance","business","gaming","sports"], "Implies the system is unfair against the viewer, igniting outrage curiosity.", "How the Stock Market Works", "Why the Stock Market Is Rigged Against You"),
    ("scam","S","Trust Violation",["personal-finance","business","tech"], "Promises exposure of a deception the viewer might be falling for.", "Is This Investment Worth It", "The Investment Scam Nobody Warns You About"),
    ("lies","S","Trust Violation",[], "Direct accusation that what they believe is false, forcing a credibility check.", "What Experts Say About Sleep", "The Lies Experts Tell You About Sleep"),
    ("doomed","S","Stakes",["history","science","personal-finance","gaming"], "Predicts inevitable failure, creating dread the viewer must resolve.", "The Future of Cities", "Why Modern Cities Are Doomed"),
    ("vanished","S","Mystery",["history","true-crime","science","travel"], "Implies an unexplained disappearance — pure curiosity gap.", "The Story of This Empire", "The Empire That Suddenly Vanished"),
    ("weaponized","S","Hidden Force",["tech","psychology","news","science"], "Reframes a neutral thing as a tool used against people.", "How Social Media Works", "How Social Media Got Weaponized Against You"),
    ("addictive","S","Hidden Force",["tech","psychology","gaming","entertainment"], "Suggests a hidden hook the viewer can't resist — they want to know why.", "Why You Use TikTok So Much", "Why TikTok Is Scientifically Addictive"),
    ("horrifying","S","Fear",["science","true-crime","history"], "Maximum emotional charge for a reveal-style payoff.", "What's In Your Tap Water", "The Horrifying Truth About Your Tap Water"),
    ("nightmare","S","Fear",["real-estate","travel","business","personal-finance"], "Frames the topic as a worst-case scenario the viewer must avoid.", "Buying Your First Home", "The First-Home Buying Nightmare Nobody Mentions"),
    ("unstoppable","S","Stakes",["business","self-improvement","sports","tech"], "Promises a force or method that can't be beaten — aspirational pull.", "Building Good Habits", "The Habit Loop That Makes You Unstoppable"),
    ("classified","S","Exclusive",["history","science","tech","news"], "Implies restricted information the viewer shouldn't have access to.", "Cold War Technology", "The Classified Cold War Tech Just Revealed"),
    ("obsessed","S","Emotional",["psychology","entertainment","fitness","culture"], "Hints at compulsion, making the behavior feel mysterious and worth dissecting.", "Why People Like True Crime", "Why We're Secretly Obsessed With True Crime"),
    ("disappeared","S","Mystery",["history","true-crime","travel","science"], "Open loop — something is gone and the viewer needs the explanation.", "This Ancient City", "The Ancient City That Disappeared Overnight"),
    ("brainwashed","S","Hidden Force",["psychology","news","culture","tech"], "Accuses an unseen force of controlling the viewer's mind.", "How Advertising Works", "How Ads Brainwashed an Entire Generation"),
    ("catastrophic","S","Stakes",["science","personal-finance","history","tech"], "Escalates a small mistake to disaster-scale consequence.", "A Common Investing Error", "The Catastrophic Investing Error You're Making"),
    ("unhinged","S","Shock",["entertainment","culture","gaming","storytelling"], "Promises chaos and unpredictability — entertainment bait.", "This Marketing Campaign", "The Most Unhinged Marketing Campaign Ever"),

    # ── A-tier: strong emotion / stakes / authority ──────────────────────────
    ("savage","A","Emotional",["sports","entertainment","gaming","business"], "Implies a brutal, no-mercy moment worth witnessing.", "How He Responded", "The Savage Way He Shut Them Down"),
    ("ruthless","A","Authority Signal",["business","history","true-crime"], "Frames a strategy or figure as coldly effective.", "How This CEO Operates", "The Ruthless Strategy This CEO Used"),
    ("lethal","A","Stakes",["fitness","science","true-crime"], "Adds danger to an otherwise ordinary subject.", "This Common Plant", "The Lethal Plant Hiding in Your Garden"),
    ("twisted","A","Mystery",["true-crime","psychology","storytelling"], "Promises a dark, unexpected turn.", "The Story Behind This Case", "The Twisted Story Behind This Case"),
    ("downfall","A","Decline Narrative",["business","history","entertainment","sports"], "Sets up a rise-and-fall arc viewers love to watch.", "What Happened to This Company", "The Downfall of a Billion-Dollar Company"),
    ("loophole","A","Exclusive",["personal-finance","business","real-estate","tech"], "Hints at a clever advantage most people don't know.", "How to Save on Taxes", "The Tax Loophole the Rich Quietly Use"),
    ("blueprint","A","Authority Signal",["business","self-improvement","fitness","real-estate"], "Promises a complete, follow-along system.", "How to Grow a Channel", "The Exact Blueprint to Grow a Channel"),
    ("relentless","A","Emotional",["self-improvement","sports","business"], "Signals intense, unstoppable effort — motivational pull.", "How to Stay Consistent", "The Relentless Routine That Builds Discipline"),
    ("meltdown","A","Dramatic Arc",["business","entertainment","personal-finance","news"], "Promises a dramatic collapse caught in real time.", "What Went Wrong Here", "The Live Meltdown Nobody Saw Coming"),
    ("predatory","A","Trust Violation",["personal-finance","business","tech"], "Frames an entity as hunting the viewer for profit.", "How These Loans Work", "The Predatory Loan Trap Targeting You"),
    ("overnight","A","Urgency",["personal-finance","business","self-improvement","fitness"], "Compresses the timeframe to feel almost too good to be true.", "How He Got Rich", "How He Got Rich Almost Overnight"),
    ("instantly","A","Urgency",["self-improvement","productivity","fitness","tech"], "Promises immediate payoff, lowering the effort barrier.", "How to Calm Anxiety", "How to Calm Anxiety Almost Instantly"),
    ("addicted","A","Hidden Force",["psychology","gaming","tech","fitness"], "Implies a compulsion worth understanding or escaping.", "Why You Crave Sugar", "Why Your Brain Is Addicted to Sugar"),
    ("collapse","A","Decline Narrative",["personal-finance","history","science","news"], "Big-stakes systemic-failure framing.", "The State of the Economy", "The Economic Collapse Everyone Ignores"),
    ("exploited","A","Trust Violation",["business","psychology","tech","personal-finance"], "Tells the viewer they're being used — outrage curiosity.", "How Free Apps Make Money", "How Free Apps Exploit Your Attention"),
    ("flawed","A","Reframe",["science","tech","education","psychology"], "Challenges something assumed to be solid or correct.", "The Science of Dieting", "The Deeply Flawed Science of Dieting"),
    ("haunting","A","Emotional",["history","true-crime","storytelling","music"], "Adds emotional weight and lingering mystery.", "The Story of This Town", "The Haunting Story This Town Buried"),
    ("rare","A","Exclusive",["automotive","photography","science","history"], "Signals scarcity, which raises perceived value.", "This Type of Car", "The Rare Car Collectors Fight Over"),
    ("genius","A","Authority Signal",["business","tech","science","gaming"], "Frames a move or design as brilliantly clever.", "How This Feature Works", "The Genius Design Trick Behind It"),
    ("brutal","A","Emotional",["fitness","sports","business","self-improvement"], "Conveys harsh intensity that signals real, not soft, content.", "The Reality of Training", "The Brutal Truth About Getting Fit"),
    ("merciless","A","Emotional",["business","sports","true-crime","history"], "Conveys cold, unrelenting intensity worth witnessing.", "How the Market Reacted", "The Merciless Way the Market Reacted"),
    ("ominous","A","Warning",["science","history","news"], "Signals a dark sign that something bad is coming.", "Signs Before the Crash", "The Ominous Sign Before the Crash"),
    ("betrayal","A","Trust Violation",["storytelling","true-crime","history","entertainment"], "Promises a turn where trust is broken — strong narrative hook.", "What His Partner Did", "The Betrayal That Ended Everything"),
    ("explosive","A","Dramatic Arc",["business","sports","entertainment","news"], "Implies sudden, high-energy escalation.", "How the Feud Started", "The Explosive Feud That Broke the Internet"),
    ("infamous","A","Dramatic Arc",["true-crime","history","entertainment","sports"], "Adds dark reputation weight that pulls clicks.", "This Famous Case", "The Infamous Case That Stunned Everyone"),

    # ── B-tier: solid curiosity / reframe / specificity ──────────────────────
    ("underrated","B","Overlooked",["gaming","tech","music","travel","cooking"], "Promises overlooked value, flattering the viewer's taste.", "Good Budget Headphones", "The Underrated Headphones Nobody Talks About"),
    ("overlooked","B","Overlooked",["business","history","science","self-improvement"], "Frames the topic as a missed opportunity worth catching.", "A Useful Productivity Tip", "The Overlooked Habit That Changes Everything"),
    ("misleading","B","Correction",["science","news","personal-finance","tech"], "Sets up a myth-busting correction.", "What the Label Says", "The Misleading Label You Trust Daily"),
    ("backfired","B","Consequence",["business","psychology","history","tech"], "Promises an ironic, instructive failure.", "Their Big Plan", "The Plan That Completely Backfired"),
    ("unexpected","B","Curiosity",[], "Primes the viewer for a surprise payoff.", "The Result of the Study", "The Unexpected Result of This Study"),
    ("counterintuitive","B","Reframe",["science","self-improvement","business","psychology"], "Signals a smart, non-obvious insight.", "How to Learn Faster", "The Counterintuitive Way to Learn Faster"),
    ("rarely","B","Exclusive",[], "Implies you're about to hear something seldom said.", "Advice About Money", "The Money Advice Rarely Given"),
    ("quietly","B","Stealth",["business","tech","news","personal-finance"], "Suggests something happening under the radar.", "What This Company Did", "What This Company Quietly Did"),
    ("accidentally","B","Discovery",["science","history","cooking","tech"], "Frames a discovery as a lucky, story-worthy mistake.", "How This Was Invented", "How This Was Accidentally Invented"),
    ("decoded","B","Disclosure",["science","tech","psychology","history"], "Promises to translate something complex into clarity.", "Understanding Body Language", "Body Language Finally Decoded"),
    ("uncovered","B","Disclosure",["news","business","history","science"], "Investigative framing that promises a reveal.", "The Real Numbers", "The Numbers They Never Uncovered"),
    ("warning","B","Warning",[], "Direct alert that primes protective attention.", "Before You Buy This", "Warning: Read This Before You Buy"),
    ("wasted","B","Consequence",["personal-finance","productivity","self-improvement"], "Triggers loss aversion about time or money.", "How You Spend Your Day", "The Hours You're Quietly Wasting"),
    ("regret","B","Emotional",["personal-finance","self-improvement","real-estate"], "Taps fear of a future mistake.", "Things to Avoid in Your 20s", "The Choice You'll Regret in Your 20s"),
    ("avoided","B","Warning",["fitness","personal-finance","cooking","travel"], "Frames the topic as a trap to dodge.", "Common Beginner Errors", "The Mistake Every Pro Avoided"),
    ("rethink","B","Reframe",["self-improvement","business","science","education"], "Invites the viewer to overturn an assumption.", "How You Manage Time", "Why You Should Rethink Time Management"),
    ("untapped","B","Overlooked",["business","real-estate","personal-finance","tech"], "Implies hidden, available opportunity.", "Side Income Ideas", "The Untapped Income Stream Nobody Uses"),
    ("notorious","B","Dramatic Arc",["true-crime","history","entertainment","gaming"], "Adds infamy and reputation weight.", "This Famous Heist", "The Notorious Heist That Fooled Everyone"),
    ("relentlessly","B","Emotional",["self-improvement","sports","business"], "Intensifier signaling sustained, serious effort.", "How to Improve Daily", "How Top Performers Relentlessly Improve"),
    ("misjudged","B","Correction",["history","psychology","business","science"], "Promises a reputation or fact being corrected.", "The Story of This Figure", "The Misjudged Genius History Forgot"),
    ("staggering","B","Specificity",["science","personal-finance","news"], "Amplifies a number or fact's scale.", "How Much Sugar You Eat", "The Staggering Amount of Sugar You Eat"),
    ("fragile","B","Stakes",["science","personal-finance","psychology","news"], "Implies something strong is secretly easy to break.", "How Stable Our System Is", "How Fragile Our System Really Is"),
    ("insider","B","Exclusive",["business","gaming","sports","tech"], "Promises norms only people on the inside know.", "How the Industry Works", "The Insider Rule Outsiders Never Hear"),

    # ── C-tier: dependable specificity / clarity boosters ────────────────────
    ("surprisingly","C","Curiosity",[], "Soft surprise cue that lifts an ordinary claim.", "This Method Works", "This Method Works Surprisingly Well"),
    ("finally","C","Urgency",[], "Implies a long-awaited answer has arrived.", "How to Fix This", "How to Finally Fix This"),
    ("simplified","C","Accessibility",["tech","science","education","personal-finance"], "Lowers the perceived effort to understand.", "Quantum Physics Explained", "Quantum Physics, Simplified"),
    ("explained","C","Accessibility",[], "Promises clarity on a confusing topic.", "The Housing Market", "The Housing Market, Explained"),
    ("really","C","Curiosity",[], "Subtle truth-seeking cue that adds intrigue.", "What This Means", "What This Really Means"),
    ("legit","C","Credibility",["fitness","self-improvement","personal-finance","tech"], "Casual credibility cue that counters skepticism.", "Is This Method Worth It", "The Only Method That's Actually Legit"),
    ("step-by-step","C","Accessibility",["education","tech","cooking","business"], "Signals an easy-to-follow process.", "How to Start Investing", "How to Start Investing, Step by Step"),
    ("ranked","C","Specificity",["gaming","entertainment","sports","tech","cooking"], "Promises a comparative, scannable list.", "Best Strategy Games", "Every Strategy Game, Ranked"),
    ("worst","C","Specificity",[], "Superlative that sets a clear, clickable stakes frame.", "Common Money Habits", "The Worst Money Habit You Have"),
    ("biggest","C","Specificity",[], "Superlative scale that promises the most important point.", "Mistakes in Cooking", "The Biggest Mistake in Cooking"),
    ("secretly","C","Stealth",[], "Implies hidden activity beneath the surface.", "What Your Phone Does", "What Your Phone Is Secretly Doing"),
    ("proven","C","Credibility",[], "Adds evidence-backed reassurance.", "A Way to Sleep Better", "The Proven Way to Sleep Better"),
]


def band(grade, i):
    arr = BANDS[grade]; return arr[i % len(arr)]


def main():
    # existing words (lowercase) to dedupe
    req = urllib.request.Request(f"{URL}/rest/v1/magnet_words?select=word", headers=H)
    existing = {r["word"].lower() for r in json.load(urllib.request.urlopen(req))}

    rows, skipped = [], []
    for i, (word, grade, cat, niches, why, before, after) in enumerate(W):
        if word.lower() in existing:
            skipped.append(word); continue
        rows.append({
            "word": word, "grade": grade, "category": cat, "lift_range": band(grade, i),
            "why_it_works": why, "example_before": before, "example_after": after,
            "niches": niches, "is_active": True,
        })

    if skipped:
        print(f"skipped {len(skipped)} dupes:", ", ".join(skipped))
    if not rows:
        print("nothing new to insert"); return

    # insert in chunks
    for c in range(0, len(rows), 50):
        chunk = rows[c:c+50]
        r = urllib.request.Request(f"{URL}/rest/v1/magnet_words", data=json.dumps(chunk).encode(), headers={**H, "Prefer": "return=minimal"}, method="POST")
        try:
            urllib.request.urlopen(r); print(f"inserted {len(chunk)}")
        except urllib.error.HTTPError as e:
            print("INSERT FAILED:", e.code, e.read().decode()[:300]); return

    total = len(json.load(urllib.request.urlopen(urllib.request.Request(f"{URL}/rest/v1/magnet_words?select=word", headers=H))))
    print(f"done — added {len(rows)} words, table now has {total}")


if __name__ == "__main__":
    main()
